pkg/apis/cloudcredential/v1/credentialsrequest_types.go

```diff
@@ -44,17 +44,55 @@ const (
 	CloudCredOperatorConfigMap = "cloud-credential-operator-config"
 )
 
-// NOTE: Run "make" to regenerate code after modifying this file
+// NOTE: Run "make update" to regenerate code after modifying this file
 
 // CredentialsRequestSpec defines the desired state of CredentialsRequest
 type CredentialsRequestSpec struct {
 	// SecretRef points to the secret where the credentials should be stored once generated.
-	SecretRef corev1.ObjectReference `json:"secretRef"`
+	// +optional
+	SecretRef corev1.ObjectReference `json:"secretRef,omitempty"`
+
+	// StorageDestination specifies the type and location of a resource to which the credentials
+	// information can be stored or linked
+	StorageDestination StorageDestination `json:"storage,omitempty"`
 
 	// ProviderSpec contains the cloud provider specific credentials specification.
 	ProviderSpec *runtime.RawExtension `json:"providerSpec,omitempty"`
 }
 
+const (
+	SecretStorageDestinationType         = "Secret"
+	ServiceAccountStorageDestinationType = "ServiceAccount"
+)
+
+// StorageDestination defines the type and location of a resource to which the credentials
+// information can be stored or linked
+// +union
+type StorageDestination struct {
+	// StorageType contains the type of storage that should be used for the credentials
+	// +unionDiscriminator
+	Type string `json:"type"`
+
+	// SecretStorageDestination specifies a secret in which to store the credentials
+	// +optional
+	Secret *SecretStorageDestination `json:"secret,omitempty"`
+
+	// SecretStorageDestination specifies a serviceaccount to which the credentials will be linked
+	// This is currently only supported on AWS
+	// +optional
+	ServiceAccount *ServiceAccountStorageDestination `json:"serviceAccount,omitempty"`
+}
+
+type SecretStorageDestination struct {
+	Namespace string `json:"namespace"`
```

Comment:
help with an example.  If I create the credential request in namespace foo, I can reference a secret in namespace bar?  Does this work today?

Reply:
I believe this works today. All component cred requests land in our openshift-cloud-credential-operator namespace by convention, but I don't think we explicitly filter on this. If you have permissions to create CredRequests, you can do what you describe.